examples/tuh_eeg_corpus.py

```diff
@@ -0,0 +1,258 @@
+    # crop recordings to tmin – tmax. can be incomplete if recording
+    # has lower duration than tmax
+    # by default mne fails if tmax is bigger than duration
+    tmax = min((raw.n_times - 1) / raw.info['sfreq'], tmax)
+    raw.crop(tmin=tmin, tmax=tmax, include_tmax=include_tmax)
+
+
+tmin = 1*60
+tmax = 6*60
+sfreq = 100
+
+preprocessors = [
+    MNEPreproc(custom_crop, tmin=tmin, tmax=tmax, include_tmax=False),
+    MNEPreproc('set_eeg_reference', ref_channels='average', ch_type='eeg'),
+    MNEPreproc(custom_rename_channels, mapping=ch_mapping),
+    MNEPreproc("pick_channels", ch_names=short_ch_names, ordered=True),
+    NumpyPreproc(lambda x: x * 1e6),
+    MNEPreproc("resample", sfreq=sfreq),
+]
+
+
+###############################################################################
+# The preprocessing loop works as follows. For every recording, we apply the
+# preprocessors as defined above. Then, we update the description of the rec,
+# since we have altered the duration, the reference, and the sampling frequency.
+# Afterwards, we split the continuous signals into compute windows. We store
+# each recording to a unique subdirectory that is named corresponding to the
+# rec id. To save memory, after windowing and storing, we delete the raw
+# dataset and the windows dataset, respectively.
+window_size_samples = 1000
+window_stride_samples = 1000
+create_compute_windows = True
+
+out_i = 0
+errors = []
+OUT_PATH = './tuh_sample/'
+tuh_splits = tuh.split([[i] for i in range(len(tuh.datasets))])
+for rec_i, tuh_subset in tuh_splits.items():
+    # implement preprocess for BaseDatasets? Would remove necessity
+    # to split above
+    preprocess(tuh_subset, preprocessors)
+
+    # update description of the recording(s)
+    tuh_subset.description.sfreq = len(tuh_subset.datasets) * [sfreq]
+    tuh_subset.description.reference = len(tuh_subset.datasets) * ['ar']
+    tuh_subset.description.n_samples = [len(d) for d in tuh_subset.datasets]
+
+    if create_compute_windows:
+        # generate compute windows here and store them to disk
+        tuh_windows = create_fixed_length_windows(
+            tuh_subset,
+            start_offset_samples=0,
+            stop_offset_samples=None,
+            window_size_samples=window_size_samples,
+            window_stride_samples=window_stride_samples,
+            drop_last_window=False
+        )
+        # save memory by deleting raw recording
+        del tuh_subset
+        # store the number of windows required for loading later on
+        tuh_windows.description["n_windows"] = [len(d) for d in
+                                                tuh_windows.datasets]
+
+        # some recordings fail with
+        # info["meas_date"] seconds must be between "(-2147483648, 0)" and "(2147483647, 0)", got "-2209161600"  # noqa
+        # during save, so catch it
+        try:
+            mne.io.meas_info._check_dates(tuh_windows.datasets[0].windows.info)
```

Comment:
using mne private functions outside mne is usually a bad idea. What's the pb with the meas_date ? what's the value that crashes?

Reply:
I cannot say. I would have to dig back into it, it's been a while since I wrote the code... Would it be possible to repair the date or make mne offer a function that returns a bool instead of rasing an error?

Reply:
I ran the example on the entire TUH EEG Corpus and checked the dates and I cannot reproduce this behavior. I suspect that something happened when I copied a tiny subset of the data to my local machine to write the code.
I will hence remove the `try` statement as well as the call to `mne.io.meas_info._check_dates`.